File: Carly_Michael/jectacular-ql/src/app/domain/ast/qls/page.ts
import {Section} from './section';
import {DefaultStyling} from './default-styling';
import {Location} from '../location';
import {QlsNode, QuestionWithAppliedStyles} from './qls-node';
import {Style} from './style';
import {QlQuestion} from '../ql/ql-question';
import {Widget} from './widget';
import {QlsVisitor} from './visitors/qls-visitor';

export class Page extends QlsNode {
  constructor(readonly name: string, readonly sections: Section[], readonly location: Location, readonly defaultSettings?: DefaultStyling) {
    super();
  }

  accept<T>(visitor: QlsVisitor<T>): T {
    return visitor.visitPage(this);
  }
}

File: DogeQL/src/main/kotlin/qls/ast/node/Page.kt
package qls.ast.node

import qls.common.Name
import qls.visitor.QlsVisitor

data class Page(val styles: List<Style>, val name: Name) : QlsNode {
    override fun <T> accept(visitor: QlsVisitor<T>): T {
        return visitor.visit(this)
    }
}